polynote-frontend/polynote/data/messages.ts

```diff
@@ -357,12 +357,46 @@ export class ExecutionStatus extends KernelStatusUpdate {
     }
 }
 
+export class Presence {
+    static codec = combined(int32, tinyStr, optional(shortStr)).to(Presence);
+    static unapply(inst: Presence): ConstructorParameters<typeof Presence> { return [inst.id, inst.name, inst.avatar]; }
+    constructor(readonly id: number, readonly name: string, readonly avatar?: string) { Object.freeze(this); }
+}
+
+export class PresenceUpdate extends KernelStatusUpdate {
+    static codec = combined(arrayCodec(uint8, Presence.codec), arrayCodec(uint8, int32)).to(PresenceUpdate);
+    static get msgTypeId() { return 5; }
+    static unapply(inst: PresenceUpdate): ConstructorParameters<typeof PresenceUpdate> {
+        return [inst.added, inst.removed];
+    }
+
+    constructor(readonly added: Presence[], readonly removed: number[]) {
+        super();
+        Object.freeze(this);
+    }
+}
+
+export class PresenceSelection extends KernelStatusUpdate {
+    static codec = combined(int32, uint16, int32, int32).to(PresenceSelection);
+    static get msgTypeId() { return 6; }
+    static unapply(inst: PresenceSelection): ConstructorParameters<typeof PresenceSelection> {
+        return [inst.presenceId, inst.cellId, inst.start, inst.length];
+    }
+
+    constructor(readonly presenceId: number, readonly cellId: number, readonly start: number, readonly length: number) {
```

Comment:
I should have used `PosRange` for this, but I didn't. I can haz some tech debt?

Reply:
Ok, but it'll ding your credit score...

Reply:
My tech debt credit score is already shot anyway...

Reply:
I guess as long as you don't want to buy a tech house any time soon you're ok... 

🤔 ... and now the metaphor is dead.

Reply:
I decided to go back and just use range. Paid off the principal before I even... ok yeah that joke is done

Reply:
Too bad there's no 😧 reaction 😉